src/pudl_archiver/zenodo/api_client.py

```diff
@@ -433,19 +260,14 @@ async def update_datapackage(self, resources: dict[str, ResourceInfo]):
             Updated Deposition.
         """
         # If nothing has changed, don't add new datapackage
-        if not self.changed():
+        if not self.changed:
             return None
 
+        if self.new_deposition is None:
```

Comment:
Curious under what circumstances this condition will be true?

Reply:
Mostly I did this because `mypy` was complaining "this could be None, so you should think twice about accessing `self.new_deposition.files`."

I think if we create a ZDI directly, instead of with `ZenodoClient.deposition_interface`, we could in call `update_datapackage` on it before setting `self.new_deposition`.